src/PrestaShopBundle/Bridge/Helper/Listing/HelperListConfigurationFactory.php

```diff
@@ -56,29 +57,34 @@ public function create(
         // @todo: investigate what these options actually do and maybe rename them to reflect it better
         bool $deleted = false,
         bool $explicitSelect = false,
-        bool $useFoundRows = true
+        bool $useFoundRows = true,
+        ?string $listId = null
     ): HelperListConfiguration {
         if (empty($defaultOrderBy)) {
             $defaultOrderBy = $identifier;
         }
-        $helperListConfiguration = new HelperListConfiguration();
 
-        $helperListConfiguration->table = $controllerConfiguration->tableName;
-        $helperListConfiguration->listId = $controllerConfiguration->tableName;
-        $helperListConfiguration->objectModelClassName = $controllerConfiguration->objectModelClassName;
-        $helperListConfiguration->identifier = $identifier;
-        $helperListConfiguration->positionIdentifier = $positionIdentifier;
-        $helperListConfiguration->isJoinLanguageTableAuto = $isJoinLanguageTableAuto;
-        $helperListConfiguration->deleted = $deleted;
-        $helperListConfiguration->defaultOrderBy = $defaultOrderBy;
-        $helperListConfiguration->explicitSelect = $explicitSelect;
-        $helperListConfiguration->useFoundRows = $useFoundRows;
-        $helperListConfiguration->id = $controllerConfiguration->tabId;
-        $helperListConfiguration->legacyControllerName = $controllerConfiguration->legacyControllerName;
-        $helperListConfiguration->token = $controllerConfiguration->token;
-        $helperListConfiguration->bootstrap = $controllerConfiguration->bootstrap;
-        $helperListConfiguration->legacyCurrentIndex = $controllerConfiguration->legacyCurrentIndex;
-        $helperListConfiguration->multishopContext = $controllerConfiguration->multiShopContext;
+        $helperListConfiguration = new HelperListConfiguration(
+            $controllerConfiguration->tabId,
+            $controllerConfiguration->tableName,
+            $listId ?: $controllerConfiguration->tableName,
+            $controllerConfiguration->objectModelClassName,
+            $identifier,
+            $positionIdentifier,
+            $isJoinLanguageTableAuto,
+            $deleted,
+            $defaultOrderBy,
+            $explicitSelect,
+            $useFoundRows,
+            $controllerConfiguration->legacyControllerName,
+            $controllerConfiguration->token,
+            $controllerConfiguration->bootstrap,
+            $controllerConfiguration->legacyCurrentIndex,
+            $controllerConfiguration->multiShopContext
+        );
+
+        $request = Request::createFromGlobals();
```

Comment:
Im pretty sure this is bad practice. What else we could do?
1. HelperListConfuguration could accept indexRoute parameter and we generate index url and use it instead. (but it would mean every time all request params will be reset)
2. inject request in configuration? (not fun to depend on request :cry: )
3.get the url from request in controller and pass it to listConfiguration? (more work in every controller when migrating)
more options?